shared/reducers/settings.js

```diff
@@ -57,197 +54,77 @@ function reducer(state: Types.State = Constants.initialState, action: SettingsGe
           unsubscribedFromAll: !name && !unsubscribedFromAll,
         },
       }
-      return {
-        ...state,
-        notifications: {
-          ...state.notifications,
+      return state.merge({
+        notifications: state.notifications.merge({
           allowEdit: false,
-          groups: {
-            ...state.notifications.groups,
-            ...changed,
-          },
-        },
-      }
+          groups: state.notifications.groups.merge(I.Map(changed)),
+        }),
+      })
```

Comment:
It seems this could also `state.update('notifications', notifications => notifications.merge...)`; same below. Is there a reason to prefer `merge` over `update`?